addons/web/static/src/views/fields/domain/domain_field.js

```diff
@@ -37,16 +38,36 @@ export class DomainField extends Component {
 
         useBus(this.env.bus, "RELATIONAL_MODEL:NEED_LOCAL_CHANGES", async (ev) => {
             if (this.isDebugEdited) {
-                const prom = this.loadCount(this.props);
+                const props = this.props;
+                const prom = this.quickValidityCheck(props);
                 ev.detail.proms.push(prom);
-                await prom;
-                if (!this.state.isValid) {
-                    this.props.record.setInvalidField(this.props.name);
-                }
+                prom.then((isValid) => {
+                    if (isValid) {
+                        this.loadCount(props);
```

Comment:
I wouldn't reload it here, because if we leave we don't care, and if we stay it will be reloaded anyway, right?